src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+        node_info (dictionary): various information associated to the nodes of the cover complex.
+    """
+    def __init__(self, input_name="data", cover_name="cover", color_name="color", verbose=False):
+        """
+        Constructor for the CoverComplexPy class.
+
+        Parameters:
+        """
+        self.input_name, self.cover_name, self.color_name, self.verbose = input_name, cover_name, color_name, verbose
+
+    def get_networkx(self, get_attrs=False):
+        """
+        Turn the 1-skeleton of the cover complex computed after calling fit() method into a networkx graph.
+        This function requires networkx (https://networkx.org/documentation/stable/install.html).
+
+        Parameters:
+            get_attrs (bool): if True, the color functions will be used as attributes for the networkx graph.
+
+        Returns:
+            G (networkx graph): graph representing the 1-skeleton of the cover complex.
+        """
+        try:
+            import networkx as nx
+            st = self.simplex_tree
+            G = nx.Graph()
+            for (splx,_) in st.get_skeleton(1):	
+                if len(splx) == 1:
+                    G.add_node(splx[0])
+                if len(splx) == 2:
+                    G.add_edge(splx[0], splx[1])
+            if get_attrs:
+                attrs = {k: {"attr_name": self.node_info[k]["colors"]} for k in G.nodes()}
+                nx.set_node_attributes(G, attrs)
+            return G
+        except ImportError:
+            print("Networkx not found, nx graph not computed")
+
+    def print_to_dot(self, epsv=.2, epss=.4):
+        """
+        Write the cover complex in a DOT file called "{self.input_name}.dot", that can be processed with, e.g., neato.
+
+        Parameters:
+            epsv (float): scale the node colors between [epsv, 1-epsv]
+            epss (float): scale the node sizes between [epss, 1-epss]
+        """
+        st = self.simplex_tree
+        node_info = self.node_info
+
+        maxv, minv = max([node_info[k]["colors"][0] for k in node_info.keys()]), min([node_info[k]["colors"][0] for k in node_info.keys()])
+        maxs, mins = max([node_info[k]["size"]      for k in node_info.keys()]), min([node_info[k]["size"]      for k in node_info.keys()])
+
+        f = open(self.input_name + ".dot", "w")
+        f.write("graph MAP{")
+        cols = []
+        for (simplex,_) in st.get_skeleton(0):
+            cnode = (1.-2*epsv) * (node_info[simplex[0]]["colors"][0] - minv)/(maxv-minv) + epsv if maxv != minv else 0
+            snode = (1.-2*epss) * (node_info[simplex[0]]["size"]-mins)/(maxs-mins) + epss if maxs != mins else 1
+            f.write(  str(simplex[0]) + "[shape=circle width=" + str(snode) + " fontcolor=black color=black label=\""  + "\" style=filled fillcolor=\"" + str(cnode) + ", 1, 1\"]")
+            cols.append(cnode)
+        for (simplex,_) in st.get_filtration():
```

Comment:
Unless the simplices need to be sorted by filtration value, we should use `get_simplices`.

Reply:
Fixed.